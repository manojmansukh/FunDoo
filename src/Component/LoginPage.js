import React, { Component } from "react";
import { View, TouchableOpacity, Text, Image, KeyboardAvoidingView, ScrollView, StyleSheet, TextInput } from "react-native";
import firebase from '../fireBase/Config';
import { onSignIn } from "./Authentication";
import { SetCurrentUser } from "./Authentication";
import { AsyncStorage } from "react-native";
import { LoginManager } from 'react-native-fbsdk';
import { getUserId } from '../Services/FireBaseDb'
import { styles } from '../CSS/LoginPage.Style';

export default class Login extends Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      emailError: null,
      passError: null,
      showPassword: true,
      hidePassword: true,
    }
  }

  managePasswordVisibility = () => {
    this.setState({ hidePassword: !this.state.hidePassword });
  }

  validateEmail = (text) => {
    let email = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
    if (email.test(text) === false) {
      this.setState({ emailError: 'Email is Not Correct' });
      return false;
    }
    else {
      //email correct
      this.setState({ email: text })
      this.setState({ emailError: null });
    }
  }

  validatePass = (text) => {
    let pass = /^[#\w@_-]{8,20}$/
    if (pass.test(text) === false) {
      this.setState({ passError: 'enter valid pass' });
      this.setState({ password: text })
      return false;
    }
    else {
      //email correct
      this.setState({ password: text })
      this.setState({ passError: null });
    }
  }

  handleLogin = () => {
    firebase.auth().signInWithEmailAndPassword(this.state.email, this.state.password)
      .then(() => {
        onSignIn();
        AsyncStorage.setItem("UserId", firebase.auth().currentUser.uid).then(getUserId())
        this.props.navigation.navigate('Lock')
        //this.props.navigation.navigate('Drawer')

      })
      .catch((error) => {
        alert("Enter the valid Details...!", error)
        var errorCode = error.code;
        var errorMessage = error.errorMessage;
      
      });
  }

  // async hadleFaceBookLogin =() => {
  //   try {
  //     let result = await LoginManager.logInWithPermissions([public_profile])
  //     if (result.isCancelled) {
  //       alert('loin was cancelled');
  //     }
  //     else{
  //       alert('Login with successful with permissdion :' 
  //       +result.grantedPermissions.toString());
  //     }

  //   } catch (error) {
  //     alert('login failed with error'+error)
  //   }
  // }
  
  render() {
    return (

      <KeyboardAvoidingView style={styles.innerView}>
        <ScrollView>
          <View style={{ flexDirection: 'row', flexWrap: "wrap", justifyContent: 'center', alignItems: 'center', top: -5 }}>

            <View style={{ justifyContent: 'center', alignItems: 'center' }}>
              <Image source={require('../Asserts/Logo.png')}
                style={{ width: 300, height: 200, top: 30, tintColor: '#1e59c7' }}
                resizeMode='contain'
              />
              <View style={{ flexDirection: 'row', alignItems: 'flex-start', top: 10, }}>
                <View ><Text style={{ color: '#4285F4', fontSize: 60, fontWeight: 'bold', }}>F</Text></View>
                <View><Text style={{ color: '#DB4437', fontSize: 60, fontWeight: 'bold' }}>u</Text></View>
                <View ><Text style={{ color: '#F4B400', fontSize: 60, fontWeight: 'bold' }}>n</Text></View>
                <View ><Text style={{ color: '#4285F4', fontSize: 60, fontWeight: 'bold' }}>D</Text></View>
                <View ><Text style={{ color: '#0F9D58', fontSize: 60, fontWeight: 'bold' }}>o</Text></View>
                <View ><Text style={{ color: '#DB4437', fontSize: 60, fontWeight: 'bold' }}>o</Text></View>
              </View>

              <Text style={styles.loginText}>
                Login
            </Text>
            </View>

            <View>
              <TextInput style={styles.input}
                testID="email"
                underlineColorAndroid="transparent"
                placeholder="Email"
                autoCapitalize="none"
                autoCorrect={false}
                keyboardType="email-address"
                onChangeText={(text) => this.validateEmail(text)} />
              <Text style={styles.error}>{this.state.emailError}</Text>

              <TextInput style={styles.input}
                testID="password"
                underlineColorAndroid="transparent"
                placeholder="Password"
                autoCapitalize="none"
                autoCorrect={false}
                secureTextEntry={this.state.hidePassword}
                onChangeText={(password) => this.setState({ password })}
                onChangeText={(text) => this.validatePass(text)} />
              <Text style={styles.error}>{this.state.passError}</Text>

              <View>
                <TouchableOpacity activeOpacity={0.8} style={styles.visibilityBtn} onPress={this.managePasswordVisibility}>
                  <Image source={(this.state.hidePassword) ? require('../Asserts/hide.png') : require('../Asserts/show.png')} style={styles.btnImage} />
                </TouchableOpacity>
              </View>

              <TouchableOpacity
                style={styles.submitButton}
                onPress={this.handleLogin}>
                <Text style={styles.submitButtonText}> Login </Text>
              </TouchableOpacity>

              <View style={{ justifyContent: 'center', alignItems: 'center', top: -9 }}>
                <Text style={{ fontWeight: 'bold' }}>OR</Text>
              </View>

              <TouchableOpacity
                // style={styles.fbButton}
                onPress={this.hadleFaceBookLogin}>
                <Image source={require('../Asserts/FbLogin.png')}
                  style={{ height: 40, width: 300, margin: 15, borderColor: 'black', borderWidth: .15, top: -15 }} />
              </TouchableOpacity>

              <TouchableOpacity onPress={() => { this.props.navigation.navigate('ForgotPass') }}>
                <Text style={styles.text}>
                  Forgot Password?
              </Text>
              </TouchableOpacity>

              <TouchableOpacity onPress={() => { this.props.navigation.navigate('SignUp') }}>
                <Text style={styles.text}>
                  Sign UP
              </Text>
              </TouchableOpacity>

            </View>

          </View>
        </ScrollView>
      </KeyboardAvoidingView>
    );
  }
}
