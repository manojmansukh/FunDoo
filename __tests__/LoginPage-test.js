import React from 'react';
import { shallow } from 'enzyme';       
import LoginPage from '../src/Component/LoginPage'

const wrapper = shallow(<LoginPage />)

describe('LoginPage', () =>{
    it('SignUpPage1 Page renders correctly', ()=>{
        expect(shallow(<LoginPage/>)).toMatchSnapshot()
    })

    it('check initial state', ()=>{
        expect(wrapper.state("email")).toEqual('');
        expect(wrapper.state("password")).toEqual('');
    })

    const emailInput = wrapper.find('TextInput').first();
    const passwordInput = wrapper.find('TextInput').last();

    it("validate email & password", ()=>{
        emailInput.props().onChangeText('manojmansukh7@gmail.com')
        passwordInput.props().onChangeText('123456')

        expect(wrapper.state("email")).toEqual('manojmansukh7@gmial.com')
        expect(wrapper.state("password")).toEqual('123456')


    })

    // it("testing email validation", ()=>{
    //     const instance= wrapper.instance();
    //     expect(instance.validateEmail()).toBe(true);

    //     emailInput.props().onChangeText('manojmansukh7@gmail.com');
    //     expect(instance.validateEmail()).toBe(true);

    //     emailInput.props().onChangeText('manojmansukh7gmail.com');
    //     expect(instance.validateEmail()).toBe(false);
    // })
})