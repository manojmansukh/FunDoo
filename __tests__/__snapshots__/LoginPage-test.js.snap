// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage SignUpPage1 Page renders correctly 1`] = `
<KeyboardAvoidingView
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <ScrollViewMock>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "top": -5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/Asserts/Logo.png",
            }
          }
          style={
            Object {
              "height": 200,
              "tintColor": "#1e59c7",
              "top": 30,
              "width": 300,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "top": 10,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#4285F4",
                  "fontSize": 60,
                  "fontWeight": "bold",
                }
              }
            >
              F
            </Text>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#DB4437",
                  "fontSize": 60,
                  "fontWeight": "bold",
                }
              }
            >
              u
            </Text>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#F4B400",
                  "fontSize": 60,
                  "fontWeight": "bold",
                }
              }
            >
              n
            </Text>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#4285F4",
                  "fontSize": 60,
                  "fontWeight": "bold",
                }
              }
            >
              D
            </Text>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#0F9D58",
                  "fontSize": 60,
                  "fontWeight": "bold",
                }
              }
            >
              o
            </Text>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#DB4437",
                  "fontSize": 60,
                  "fontWeight": "bold",
                }
              }
            >
              o
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "display": "flex",
              "fontSize": 40,
              "fontWeight": "bold",
              "marginBottom": 10,
              "top": -2,
            }
          }
        >
          Login
        </Text>
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Email"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "borderColor": "#314bb0",
              "borderWidth": 1,
              "height": 40,
              "marginTop": 10,
              "width": "90%",
            }
          }
          testID="email"
          underlineColorAndroid="transparent"
        />
        <Text
          style={
            Object {
              "color": "red",
              "flexDirection": "row",
              "fontSize": 11,
              "marginHorizontal": 30,
              "marginTop": 1,
              "width": "90%",
            }
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="Password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "alignSelf": "center",
              "borderColor": "#314bb0",
              "borderWidth": 1,
              "height": 40,
              "marginTop": 10,
              "width": "90%",
            }
          }
          testID="password"
          underlineColorAndroid="transparent"
        />
        <Text
          style={
            Object {
              "color": "red",
              "flexDirection": "row",
              "fontSize": 11,
              "marginHorizontal": 30,
              "marginTop": 1,
              "width": "90%",
            }
          }
        />
        <View>
          <TouchableOpacity
            activeOpacity={0.8}
            onPress={[Function]}
            style={
              Object {
                "bottom": -55,
                "height": 190,
                "padding": 5,
                "position": "absolute",
                "right": 20,
                "width": 35,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/Asserts/hide.png",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "resizeMode": "contain",
                  "width": "100%",
                }
              }
            />
          </TouchableOpacity>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#1e86c7",
              "fontSize": 50,
              "height": 40,
              "margin": 15,
              "top": -2,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 15,
                "padding": 10,
              }
            }
          >
             Login 
          </Text>
        </TouchableOpacity>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "top": -9,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            OR
          </Text>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/Asserts/FbLogin.png",
              }
            }
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 0.15,
                "height": 40,
                "margin": 15,
                "top": -15,
                "width": 300,
              }
            }
          />
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "flexDirection": "row",
                "fontSize": 15,
                "top": -15,
              }
            }
          >
            Forgot Password?
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "flexDirection": "row",
                "fontSize": 15,
                "top": -15,
              }
            }
          >
            Sign UP
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </ScrollViewMock>
</KeyboardAvoidingView>
`;
